### PR TITLE
Sharethrough: Add support for ID5, Shared ID, and Live Intent ID

### PR DESCRIPTION
**Pull request by poojawins**
## Type of change
- [x] Feature
- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?

## Description of change
- Add support for ID5, Shared ID, and Live Intent ID
- PR on the docs repo: prebid/prebid.github.io#2681
pubgrowth.engineering@sharethrough.com

### CODE REVIEW

File: modules/sharethroughBidAdapter.js [hunk 1]
```diff
@@ -147,6 +135,34 @@ export const sharethroughAdapterSpec = {
...
+  if (ttd) universalIds.ttduid = ttd;
+
+  const pubc = utils.deepAccess(bidRequest, 'userId.pubcid') || utils.deepAccess(bidRequest, 'crumbs.pubcid');
+  if (pubc) universalIds.pubcid = pubc;
+
+  const idl = utils.deepAccess(bidRequest, 'userId.idl_env');
+  if (idl) universalIds.idluid = idl;
+
+  const id5 = utils.deepAccess(bidRequest, 'userId.id5id');
+  if (id5) universalIds.id5uid = id5;
```
Comment (smenzer) [inline modules/sharethroughBidAdapter.js:153]: just as an fyi, the `userId.id5id` is an object, not a string. the object looks like this:
```javascript
{
  uid: 'ID5-12345', // this is the ID5 Universal ID
  ext: {
    linkType: 2 // possible options are 0, 1, or 2
  }
}
```

It is highly recommended that you pass through the entire object to your servers so you can pass through the `linkType` to DSPs, several of whom have asked for this field. If you have any questions, feel free to reach out to me directly (scott at id5 dot io)

File: test/spec/modules/sharethroughBidAdapter_spec.js [hunk 2]
```diff
@@ -15,7 +16,12 @@ const bidRequests = [
     userId: {
       tdid: 'fake-tdid',
       pubcid: 'fake-pubcid',
-      idl_env: 'fake-identity-link'
+      idl_env: 'fake-identity-link',
+      id5id: 'fake-id5id',
```
Comment (smenzer) [inline test/spec/modules/sharethroughBidAdapter_spec.js:20]: as noted above, this is not what the `id5id` looks like, so you'll want to update it to ensure your tests are working properly.

**Review by smenzer — CHANGES_REQUESTED** [review]
see comments inline about the ID5 ID

File: modules/sharethroughBidAdapter.js (hunk 1 above)
Comment (maphe) [reply modules/sharethroughBidAdapter.js:153]: @smenzer thanks for raising that, we missed it and I'm gonna push a fix soon. Is there a doc somewhere I can link internally to describe the meaning of `linkType` and its possible values?

Comment (smenzer) [reply modules/sharethroughBidAdapter.js]: @maphe great to hear! we have some docs on our wiki - https://wiki.id5.io/x/EABH - but basically, `linkType` can be 0 (not linked across domains), 1 (linked across domains with probabilistic signals), or 2 (linked across domains with deterministic signals)

**Comment by maphe** [comment]
@smenzer thanks for the link, that's perfect. Also I addressed your comments 👍 feel free to take another look

File: modules/sharethroughBidAdapter.js (hunk 1 above)
Comment (smenzer) [reply modules/sharethroughBidAdapter.js]: > @smenzer thanks for the link, that's perfect. Also I addressed your comments 👍 feel free to take another look

Looks great!

**Review by smenzer — APPROVED** [review]
I've approved the update to the ID5 code specifically, I didn't really review the rest of the change, though

**Review by osazos — APPROVED** [review]
LGTM

**Pull request merged**